Fix: add licenses for bundled luarocks modules

SlySven opened the pull request:
When we make an installable or distributable build we compile some files from various Luarocks and as they are under the same license as Lua (MIT) we MUST mention them and their license in the 3rd party components tab in the "About Mudlet" dialogue for those builds that we distribute from the https://www.mudlet.org site. However users building Mudlet personally are more likely to source the luarocks themselves and as such it is not relevant to mention these things. Setting the environmental variable `WITH_BUNDLED_LUA_FILES` to `NO` will, as a result of this PR suppress mention of them from the 3rd party component tab but otherwise they will show as required.

I have also taken the opportunity to clean up some other details in the "3rd Party" tab in the "About Mudlet" dialogue - many some initial capital letters and formatting of some included URLS. Also a typo of `<ht>` instead of `<hl>`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: add licenses for bundled luarocks modules

When we make an installable or distributable build we compile some files
from various Luarocks and as they are under the same license as Lua (MIT)
we MUST mention them and their license in the 3rd party components tab in
the "About Mudlet" dialog for those builds that we distribute from the
https://www.mudlet.org site. However users building Mudlet personally are
more likely to source the luarocks themselves and as such it is not
relevant to mention these things. Setting the environmental variable
`WITH_BUNDLED_LUA_FILES` to `NO` will, as a result of this PR suppress
mention of them from the 3rd party component tab but otherwise they will
show as required.

I have also taken the oportunity to clean up some other details in the
"3rd Party" tab in the "About Mudlet" dialogue - many some initial capital
letters and formatting of some included URLS. Also a typo of `<ht>` instead
of `<hl>`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (commented):
Looks good but let's not add a dedicated option for it. The harm for leaving the licence notices in is small, but the harm for adding more an more options for trivial things - blowing up the amount of them - is more.

SlySven commented:
> Looks good but let's not add a dedicated option for it. The harm for leaving the licence notices in is small, but the harm for adding more an more options for trivial things - blowing up the amount of them - is more.

Yeah, but it is not about an option for trivial things - there is absolutely no need to include the 3rd party licences details for things that **we do not supply for those who build Mudlet for themselves** (because they source those third party things themselves and we have **NO** hand in distributing them) - so they (self-builders) can use `WITH_BUNDLED_LUA_FILES="NO"`. This helps to keep the tab from including some highly repetitive stuff that it doesn't need.

For everyone else, including:
* those packaging up a Mudlet build to distribute as a product for a Linux (or other OS, including me when I work out the details for a MSYS2/Mingw-w64 "PGKBUILD" or a FreeBSD "port") distribution
* ourselves as Mudlet Makers making a stand-alone installable product

then the default case will be include (and that is a *mandatory* term of the licence under which we use those licenced "bits")  those extra 5 or so identical copies of the MIT licence with a different title for each "bit".

I do intend to add a section to the "Compiling Mudlet" part of the Wiki about all the "build" `WITH_XXX` options and under which circumstances a builder might want/need/like to set any of them to the non default value of `NO`.

vadi2 commented:
We've got to be very careful not to bloat the client with the options we _think_ people need, but in practice nobody is actually asking for. It's too easy to get carried away!

SlySven committed:
Revise: include the licenses for the Luarock modules even when NOT required

This was requested during peer-review and it doesn't make sense to me - why
report about items that are not inherently in the source code and which
we do not distribute with that - when the only reason we do have to mention
them is when we ship their binaries as part of a precompiled build? OTOH
it has been pointed out that it does remove one "option" that self-buildera
could frob when they build Mudlet from source so will make things
marginally simpler for them.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
